src/main/java/games/strategy/engine/framework/startup/ui/PBEMSetupPanel.java

```diff
@@ -259,17 +257,10 @@ private void loadEmailSender(final GameData data) {
    *        the type of class
    * @return a IBean either loaded from the cache or created
    */
-  private static <T extends IBean> T findCachedOrCreateNew(final Class<T> theClassType) {
-    T cached = theClassType.cast(LocalBeanCache.INSTANCE.getSerializable(theClassType.getCanonicalName()));
-    if (cached == null) {
-      try {
-        cached = theClassType.getDeclaredConstructor().newInstance();
-      } catch (final Exception e) {
-        throw new RuntimeException(
-            "Bean of type " + theClassType + " doesn't have public default constructor, error: " + e.getMessage());
-      }
-    }
-    return cached;
+  private static <T extends IBean> T useCacheIfAvailable(final T instance) {
```

Comment:
Nit: Would recommend using a `Supplier<T>` here in order to lazily create the instance in the event it's already cached.  That could be important if creating a `T` instance has any side effects (I don't think it does currently).  Then you can just pass a method reference where this method is called (e.g. `NullEmailSender::new` instead of `new NullEmailSender()`).

Reply:
But how would we get the class name then?

Reply:
Yes, you're right.  I missed that little detail. :smile:

You'd have to then pass two parameters: a `Class<T>` _and_ a `Supplier<T>`.  Not sure if you want to do that, but it does seem a bit strange to create an instance just to use its `Class` so you can look up a cached instance of the same type.  But not a big deal if you leave it as-is.